zram to be built in CI

ZRAM service is not installing in CI due to missing gcc, see
```
gcc -Wall -std=c99 -c main.c
make: gcc: Command not found
make: *** [makefile:11: main.o] Error 127
install: cannot stat 'overlayfs-tools/overlay': No such file or directory
#####          Reboot to activate zram-config         #####
#####       edit /etc/ztab to configure options       #####
zram-config: unrecognized service
```
https://travis-ci.com/github/openhab/openhabian/jobs/342255583

Please add all tools required to compile and install it to dockerfiles.

--- a/.travis.yml
+++ b/.travis.yml
@@ -20,8 +20,8 @@ services:
   - docker
 
 stages:
-  - Dev & Unit tests
-  - BATS and shellcheck tests
+        #  - Dev & Unit tests
+        #  - BATS and shellcheck tests
   - Run unattended openHABian installation
 
 install:

--- a/.travis.yml
+++ b/.travis.yml
@@ -26,11 +26,11 @@ stages:
 
 install:
   # prepare configuration for tests, select debug level here:
-  - sed -iE 's#mode=.*$#mode=unattended_debug#' build-image/openhabian.conf
-  - #sed -iE 's#mode=.*$#mode=debug_maximum#' build-image/openhabian.conf
+  - sed -iE 's#mode=.*$#debugmode=on#' build-image/openhabian.conf
+  - #sed -iE 's#mode=.*$#debugmode=maximum#' build-image/openhabian.conf
   - echo "Cloning openHABian repo from https://github.com/${TRAVIS_REPO_SLUG}.git"
   - sed -i 's#repositoryurl=https://github.com/openhab/openhabian.git#repositoryurl=https://github.com/'${TRAVIS_REPO_SLUG}'.git#' build-image/openhabian.conf
-  - sed -i 's#clonebranch=master#clonebranch='${TRAVIS_BRANCH}'#' build-image/openhabian.conf
+  - sed -i 's#clonebranch=.*$#clonebranch='${TRAVIS_BRANCH}'#' build-image/openhabian.conf
 
 jobs:
   fast_finish: true

--- a/.travis.yml
+++ b/.travis.yml
@@ -21,8 +21,8 @@ services:
   - docker
 
 stages:
-        #  - Dev & Unit tests
-        #  - BATS and shellcheck tests
+  - Dev & Unit tests
+  - BATS and shellcheck tests
   - Run unattended openHABian installation
 
 install:

--- a/.travis.yml
+++ b/.travis.yml
@@ -46,7 +46,7 @@ jobs:
         - sed -i 's/^java_opt=.*/java_opt=Zulu8-64/' build-image/openhabian.conf
         - docker build -t openhabian/dev-openhabian -f Dockerfile.openhabian-aarch64-native .
         - docker run -i openhabian/dev-openhabian bash -c '/usr/local/bin/bats -r -f "dev-." .'
-        - docker run -i openhabian/dev-openhabian bash -c '/usr/local/bin/bats -r -f "unit-." .'
+        - travis_wait 15 docker run -i openhabian/dev-openhabian bash -c '/usr/local/bin/bats -r -f "unit-." .'
     - stage: BATS and shellcheck tests
       env: aarch64
       arch: arm64

--- a/.travis.yml
+++ b/.travis.yml
@@ -46,7 +46,7 @@ jobs:
         - sed -i 's/^java_opt=.*/java_opt=Zulu8-64/' build-image/openhabian.conf
         - docker build -t openhabian/dev-openhabian -f Dockerfile.openhabian-aarch64-native .
         - docker run -i openhabian/dev-openhabian bash -c '/usr/local/bin/bats -r -f "dev-." .'
-        - travis_wait 15 docker run -i openhabian/dev-openhabian bash -c '/usr/local/bin/bats -r -f "unit-." .'
+        - travis_wait 30 docker run -i openhabian/dev-openhabian bash -c '/usr/local/bin/bats -r -f "unit-." .'
     - stage: BATS and shellcheck tests
       env: aarch64
       arch: arm64

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,8 @@
 language: shell
 os: linux
 # using latest Ubuntu (not the travis-ci default) to get latest tool versions in Travis environment
-dist: bionic
+#dist: bionic
+dist: focal
 
 cache: false
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -46,7 +46,7 @@ jobs:
         - sed -i 's/^java_opt=.*/java_opt=Zulu8-64/' build-image/openhabian.conf
         - docker build -t openhabian/dev-openhabian -f Dockerfile.openhabian-aarch64-native .
         - docker run -i openhabian/dev-openhabian bash -c '/usr/local/bin/bats -r -f "dev-." .'
-        - travis_wait 30 docker run -i openhabian/dev-openhabian bash -c '/usr/local/bin/bats -r -f "unit-." .'
+        - docker run -i openhabian/dev-openhabian bash -c '/usr/local/bin/bats -r -f "unit-." .'
     - stage: BATS and shellcheck tests
       env: aarch64
       arch: arm64

--- a/Dockerfile.openhabian-ARMv6hf-emulated
+++ b/Dockerfile.openhabian-ARMv6hf-emulated
@@ -9,6 +9,7 @@ ENV APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=DontWarn
 
 RUN apt-get update -qq \
     && apt-get install --yes -qq --no-install-recommends systemd systemd-sysv git wget python3 python3-pip apt-utils jq virt-what \
+       lsb-release unzip apt-transport-https \
     && rm -rf /var/lib/apt/lists/*
 RUN git clone https://github.com/bats-core/bats-core.git && \
     cd bats-core && \
@@ -38,8 +39,10 @@ RUN cp docker-systemctl-replacement/files/docker/systemctl3.py /bin/systemctl
 
 ENTRYPOINT ["/usr/bin/entry.sh"]
 
-RUN adduser openhabian --gecos "Openhabian,,," --disabled-password
+RUN adduser openhabian --gecos "openHABian,,," --disabled-password
+RUN adduser openhab --gecos "openHABian,,," --disabled-password
 RUN echo "openhabian:openhabian" | chpasswd
+RUN echo "openhab:openhabian" | chpasswd
 RUN /bin/echo -n "Running on " && /usr/bin/arch
 
 COPY . /opt/openhabian/

--- a/Dockerfile.openhabian-ARMv6hf-emulated
+++ b/Dockerfile.openhabian-ARMv6hf-emulated
@@ -1,5 +1,5 @@
 # RPi1, RPi0W (ARMv6hf arch)
-FROM balenalib/raspberry-pi-debian-node:latest-buster
+FROM balenalib/raspberry-pi-debian-node:latest-buster-build
 
 RUN [ "cross-build-start" ]
 

--- a/Dockerfile.openhabian-ARMv6hf-native
+++ b/Dockerfile.openhabian-ARMv6hf-native
@@ -8,6 +8,7 @@ ENV APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=DontWarn
 
 RUN apt-get update -qq \
     && apt-get install --yes -qq --no-install-recommends systemd systemd-sysv git wget python3 python3-pip apt-utils jq virt-what \
+       lsb-release unzip apt-transport-https \
     && rm -rf /var/lib/apt/lists/*
 RUN git clone https://github.com/bats-core/bats-core.git && \
     cd bats-core && \
@@ -37,8 +38,10 @@ RUN cp docker-systemctl-replacement/files/docker/systemctl3.py /bin/systemctl
 
 ENTRYPOINT ["/usr/bin/entry.sh"]
 
-RUN adduser openhabian --gecos "Openhabian,,," --disabled-password
+RUN adduser openhabian --gecos "openHABian,,," --disabled-password
+RUN adduser openhab --gecos "openHABian,,," --disabled-password
 RUN echo "openhabian:openhabian" | chpasswd
+RUN echo "openhab:openhabian" | chpasswd
 RUN /bin/echo -n "Running on " && /usr/bin/arch
 
 COPY . /opt/openhabian/

--- a/Dockerfile.openhabian-ARMv6hf-native
+++ b/Dockerfile.openhabian-ARMv6hf-native
@@ -1,5 +1,5 @@
 # RPi1, RPi0W (ARMv6hf arch)
-FROM balenalib/raspberry-pi-debian-node:latest-buster
+FROM balenalib/raspberry-pi-debian-node:latest-buster-build
 
 
 ENV container docker

--- a/Dockerfile.openhabian-aarch64-64bit
+++ b/Dockerfile.openhabian-aarch64-64bit
@@ -1,7 +1,7 @@
 # Note RPi3 base (first RPi to have aarch64 available) WITH 64bit OS
 #FROM balenalib/raspberrypi3-debian-node:latest-buster
 #FROM balenalib/generic-aarch64-debian-node
-FROM balenalib/raspberrypi4-64-debian-node
+FROM balenalib/raspberrypi4-64-debian-node-build
 
 
 ENV container docker

--- a/Dockerfile.openhabian-aarch64-64bit
+++ b/Dockerfile.openhabian-aarch64-64bit
@@ -10,6 +10,7 @@ ENV APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=DontWarn
 
 RUN apt-get update -qq \
     && apt-get install --yes -qq --no-install-recommends systemd systemd-sysv git wget python3 python3-pip apt-utils jq virt-what \
+       lsb-release unzip apt-transport-https \
     && rm -rf /var/lib/apt/lists/*
 RUN git clone https://github.com/bats-core/bats-core.git && \
     cd bats-core && \
@@ -39,8 +40,10 @@ RUN cp docker-systemctl-replacement/files/docker/systemctl3.py /bin/systemctl
 
 ENTRYPOINT ["/usr/bin/entry.sh"]
 
-RUN adduser openhabian --gecos "Openhabian,,," --disabled-password
+RUN adduser openhabian --gecos "openHABian,,," --disabled-password
+RUN adduser openhab --gecos "openHABian,,," --disabled-password
 RUN echo "openhabian:openhabian" | chpasswd
+RUN echo "openhab:openhabian" | chpasswd
 RUN /bin/echo -n "Running on " && /usr/bin/arch
 
 COPY . /opt/openhabian/

--- a/Dockerfile.openhabian-aarch64-native
+++ b/Dockerfile.openhabian-aarch64-native
@@ -14,7 +14,8 @@ ENV APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=DontWarn
 # - unzip (java)
 # those packages would normally be included in our standard install
 RUN apt-get update -qq \
-    && apt-get install --yes -qq --no-install-recommends systemd systemd-sysv git wget python3 python3-pip apt-utils jq virt-what lsb-release unzip psmisc \
+    && apt-get install --yes -qq --no-install-recommends systemd systemd-sysv git wget python3 python3-pip apt-utils jq virt-what \
+       lsb-release unzip apt-transport-https \
     && rm -rf /var/lib/apt/lists/*
 RUN git clone https://github.com/bats-core/bats-core.git && \
     cd bats-core && \
@@ -44,8 +45,10 @@ RUN cp docker-systemctl-replacement/files/docker/systemctl3.py /bin/systemctl
 
 ENTRYPOINT ["/usr/bin/entry.sh"]
 
-RUN adduser openhabian --gecos "Openhabian,,," --disabled-password
+RUN adduser openhabian --gecos "openHABian,,," --disabled-password
+RUN adduser openhab --gecos "openHABian,,," --disabled-password
 RUN echo "openhabian:openhabian" | chpasswd
+RUN echo "openhab:openhabian" | chpasswd
 RUN /bin/echo -n "Running on " && /usr/bin/arch
 
 COPY . /opt/openhabian/

--- a/Dockerfile.openhabian-aarch64-native
+++ b/Dockerfile.openhabian-aarch64-native
@@ -1,7 +1,6 @@
 # Note RPi3 base (first RPi to have aarch64 available)
 # ATTENTION: Zulu Java on raspberrypi3 images fails to run
-FROM balenalib/raspberrypi3-debian-node:latest-buster
-#FROM balenalib/generic-aarch64-debian-node
+FROM balenalib/raspberrypi3-debian-node:latest-buster-build
 
 
 ENV container docker

--- a/Dockerfile.openhabian-arm32hf-CM3
+++ b/Dockerfile.openhabian-arm32hf-CM3
@@ -1,6 +1,6 @@
 # CM3 is like RPi3 but without BT,WiFi,Ethernet
 # ATTENTION: Zulu Java on raspberrypi3 images fails to run
-FROM balenalib/fincm3-debian-node:latest-buster
+FROM balenalib/fincm3-debian-node:latest-buster-build
 
 
 ENV container docker

--- a/Dockerfile.openhabian-arm32hf-CM3
+++ b/Dockerfile.openhabian-arm32hf-CM3
@@ -9,6 +9,7 @@ ENV APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=DontWarn
 
 RUN apt-get update -qq \
     && apt-get install --yes -qq --no-install-recommends systemd systemd-sysv git wget python3 python3-pip apt-utils jq virt-what \
+       lsb-release unzip apt-transport-https \
     && rm -rf /var/lib/apt/lists/*
 RUN git clone https://github.com/bats-core/bats-core.git && \
     cd bats-core && \
@@ -38,8 +39,10 @@ RUN cp docker-systemctl-replacement/files/docker/systemctl3.py /bin/systemctl
 
 ENTRYPOINT ["/usr/bin/entry.sh"]
 
-RUN adduser openhabian --gecos "Openhabian,,," --disabled-password
+RUN adduser openhabian --gecos "openHABian,,," --disabled-password
+RUN adduser openhab --gecos "openHABian,,," --disabled-password
 RUN echo "openhabian:openhabian" | chpasswd
+RUN echo "openhab:openhabian" | chpasswd
 RUN /bin/echo -n "Running on " && /usr/bin/arch
 
 COPY . /opt/openhabian/

--- a/Dockerfile.openhabian-arm32hf-emulated
+++ b/Dockerfile.openhabian-arm32hf-emulated
@@ -9,6 +9,7 @@ ENV APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=DontWarn
 
 RUN apt-get update -qq \
     && apt-get install --yes -qq --no-install-recommends systemd systemd-sysv git wget python3 python3-pip apt-utils jq virt-what \
+       lsb-release unzip apt-transport-https \
     && rm -rf /var/lib/apt/lists/*
 RUN git clone https://github.com/bats-core/bats-core.git && \
     cd bats-core && \
@@ -38,8 +39,10 @@ RUN cp docker-systemctl-replacement/files/docker/systemctl3.py /bin/systemctl
 
 ENTRYPOINT ["/usr/bin/entry.sh"]
 
-RUN adduser openhabian --gecos "Openhabian,,," --disabled-password
+RUN adduser openhabian --gecos "openHABian,,," --disabled-password
+RUN adduser openhab --gecos "openHABian,,," --disabled-password
 RUN echo "openhabian:openhabian" | chpasswd
+RUN echo "openhab:openhabian" | chpasswd
 RUN /bin/echo -n "Running on " && /usr/bin/arch
 
 COPY . /opt/openhabian/

--- a/Dockerfile.openhabian-arm32hf-emulated
+++ b/Dockerfile.openhabian-arm32hf-emulated
@@ -1,5 +1,5 @@
 # assumes at least ARMv71 but not ARMv8 (aarch64) yet
-FROM balenalib/raspberry-pi2-debian-node:latest-buster
+FROM balenalib/raspberry-pi2-debian-node:latest-buster-build
 
 RUN [ "cross-build-start" ]
 

--- a/Dockerfile.openhabian-arm32hf-native
+++ b/Dockerfile.openhabian-arm32hf-native
@@ -8,6 +8,7 @@ ENV APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=DontWarn
 
 RUN apt-get update -qq \
     && apt-get install --yes -qq --no-install-recommends systemd systemd-sysv git wget python3 python3-pip apt-utils jq virt-what \
+       lsb-release unzip apt-transport-https \
     && rm -rf /var/lib/apt/lists/*
 RUN git clone https://github.com/bats-core/bats-core.git && \
     cd bats-core && \
@@ -37,8 +38,10 @@ RUN cp docker-systemctl-replacement/files/docker/systemctl3.py /bin/systemctl
 
 ENTRYPOINT ["/usr/bin/entry.sh"]
 
-RUN adduser openhabian --gecos "Openhabian,,," --disabled-password
+RUN adduser openhabian --gecos "openHABian,,," --disabled-password
+RUN adduser openhab --gecos "openHABian,,," --disabled-password
 RUN echo "openhabian:openhabian" | chpasswd
+RUN echo "openhab:openhabian" | chpasswd
 RUN /bin/echo -n "Running on " && /usr/bin/arch
 
 COPY . /opt/openhabian/

--- a/Dockerfile.openhabian-arm32hf-native
+++ b/Dockerfile.openhabian-arm32hf-native
@@ -1,5 +1,5 @@
 # assumes at least ARMv71 but not ARMv8 (aarch64) yet
-FROM balenalib/raspberry-pi2-debian-node:latest-buster
+FROM balenalib/raspberry-pi2-debian-node:latest-buster-build
 
 
 ENV container docker

--- a/Dockerfile.openhabian-x86
+++ b/Dockerfile.openhabian-x86
@@ -7,6 +7,7 @@ ENV APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=DontWarn
 
 RUN apt-get update -qq \
     && apt-get install --yes -qq --no-install-recommends systemd systemd-sysv git wget python3 python3-pip apt-utils jq virt-what \
+       lsb-release unzip apt-transport-https \
     && rm -rf /var/lib/apt/lists/*
 RUN git clone https://github.com/bats-core/bats-core.git && \
     cd bats-core && \
@@ -36,8 +37,10 @@ RUN cp docker-systemctl-replacement/files/docker/systemctl3.py /bin/systemctl
 
 ENTRYPOINT ["/usr/bin/entry.sh"]
 
-RUN adduser openhabian --gecos "Openhabian,,," --disabled-password
+RUN adduser openhabian --gecos "openHABian,,," --disabled-password
+RUN adduser openhab --gecos "openHABian,,," --disabled-password
 RUN echo "openhabian:openhabian" | chpasswd
+RUN echo "openhab:openhabian" | chpasswd
 RUN /bin/echo -n "Running on " && /usr/bin/arch
 
 COPY . /opt/openhabian/

--- a/Dockerfile.openhabian-x86
+++ b/Dockerfile.openhabian-x86
@@ -1,4 +1,4 @@
-FROM balenalib/intel-nuc-debian-node:latest-buster
+FROM balenalib/intel-nuc-debian-node:latest-buster-build
 
 
 ENV container docker

--- a/build-image/first-boot.bash
+++ b/build-image/first-boot.bash
@@ -162,7 +162,8 @@ if [ -d /opt/openhabian/ ]; then cd /opt && rm -rf /opt/openhabian/; fi
 # shellcheck disable=SC2154
 set -x
 echo "$(timestamp) [openHABian] Cloning myself from ${repositoryurl}, ${clonebranch} branch... "
-if git clone -q -b "$clonebranch" "$repositoryurl" /opt/openhabian; then echo "OK"; else echo "FAILED"; fail_inprogress; fi
+if ! git clone -q -b "$clonebranch" "$repositoryurl" /opt/openhabian; then echo "FAILED"; fail_inprogress; fi
+echo "OK"
 ln -sfn /opt/openhabian/openhabian-setup.sh /usr/local/bin/openhabian-config
 
 # shellcheck disable=SC2154

--- a/build-image/first-boot.bash
+++ b/build-image/first-boot.bash
@@ -158,9 +158,9 @@ if hash python3 2>/dev/null; then bash /boot/webif.bash reinsure_running; fi
 echo -n "$(timestamp) [openHABian] Installing git package... "
 if apt-get install --yes git &>/dev/null; then echo "OK"; else echo "FAILED"; fail_inprogress; fi
 
-if [ -d /opt/openhabian/ ]; then cd /opt && rm -rf /opt/openhabian/; fi
-# shellcheck disable=SC2154
 set -x
+if [ -d /opt/openhabian ]; then cd /opt && rm -rf /opt/openhabian; fi
+# shellcheck disable=SC2154
 echo "$(timestamp) [openHABian] Cloning myself from ${repositoryurl}, ${clonebranch} branch... "
 if ! git clone -q -b "$clonebranch" "$repositoryurl" /opt/openhabian; then echo "FAILED"; fail_inprogress; fi
 echo "OK"

--- a/build-image/first-boot.bash
+++ b/build-image/first-boot.bash
@@ -178,8 +178,8 @@ echo "$(timestamp) [openHABian] Execution of 'openhabian-setup.sh unattended' co
 echo -n "$(timestamp) [openHABian] Waiting for openHAB to become ready on $HOSTNAME ..."
 
 # this took ~130 seconds on a RPi2
-if tryUntil "wget -S --spider -t 3 --waitretry=4 http://${HOSTNAME}:8080/start/index 2>&1 | grep -q 'HTTP/1.1 200 OK'" 20 10; then echo "failed."; exit 1; fi
-echo " OK"
+if tryUntil "wget -S --spider -t 3 --waitretry=4 http://${HOSTNAME}:8080/start/index 2>&1 | grep -q 'HTTP/1.1 200 OK'" 20 10; then echo "FAILED"; exit 1; fi
+echo "OK"
 
 echo "$(timestamp) [openHABian] Visit the openHAB dashboard now: http://${HOSTNAME:-openhab}:8080"
 echo "$(timestamp) [openHABian] To gain access to a console, simply reconnect."

--- a/build-image/first-boot.bash
+++ b/build-image/first-boot.bash
@@ -190,8 +190,9 @@ sleep 12
 if hash python3 2>/dev/null; then bash /boot/webif.bash cleanup; fi
 
 if [ -z "$SILENT" ]; then
-  echo -e "\\n\\e[36mMemory usage:"
+  echo -e "\\n${COL_CYAN}Memory usage:"
   free -m && ps -auxq "$(cat /var/lib/openhab2/tmp/karaf.pid)" |awk '/openhab/ {print "size/res="$5"/"$6" KB"}'
+  echo "${COL_DEF}"
 fi
 
 # vim: filetype=sh

--- a/build-image/first-boot.bash
+++ b/build-image/first-boot.bash
@@ -160,7 +160,7 @@ if apt-get install --yes git &>/dev/null; then echo "OK"; else echo "FAILED"; fa
 
 if [ -d /opt/openhabian/ ]; then cd /opt && rm -rf /opt/openhabian/; fi
 # shellcheck disable=SC2154
-echo -n "$(timestamp) [openHABian] Cloning myself from ${repositoryurl}, ${clonebranch} branch... "
+echo "$(timestamp) [openHABian] Cloning myself from ${repositoryurl}, ${clonebranch} branch... "
 if git clone -q -b "$clonebranch" "$repositoryurl" /opt/openhabian; then echo "OK"; else echo "FAILED"; fail_inprogress; fi
 ln -sfn /opt/openhabian/openhabian-setup.sh /usr/local/bin/openhabian-config
 

--- a/build-image/first-boot.bash
+++ b/build-image/first-boot.bash
@@ -160,7 +160,7 @@ if apt-get install --yes git &>/dev/null; then echo "OK"; else echo "FAILED"; fa
 
 if [ -d /opt/openhabian ]; then cd /opt && rm -rf /opt/openhabian; fi
 # shellcheck disable=SC2154
-echo "$(timestamp) [openHABian] Cloning myself from ${repositoryurl}, ${clonebranch} branch... "
+echo -n "$(timestamp) [openHABian] Cloning myself from ${repositoryurl}, ${clonebranch} branch... "
 if ! git clone -q -b "$clonebranch" "$repositoryurl" /opt/openhabian; then echo "FAILED"; fail_inprogress; fi
 echo "OK"
 ln -sfn /opt/openhabian/openhabian-setup.sh /usr/local/bin/openhabian-config

--- a/build-image/first-boot.bash
+++ b/build-image/first-boot.bash
@@ -158,7 +158,6 @@ if hash python3 2>/dev/null; then bash /boot/webif.bash reinsure_running; fi
 echo -n "$(timestamp) [openHABian] Installing git package... "
 if apt-get install --yes git &>/dev/null; then echo "OK"; else echo "FAILED"; fail_inprogress; fi
 
-set -x
 if [ -d /opt/openhabian ]; then cd /opt && rm -rf /opt/openhabian; fi
 # shellcheck disable=SC2154
 echo "$(timestamp) [openHABian] Cloning myself from ${repositoryurl}, ${clonebranch} branch... "
@@ -168,7 +167,6 @@ ln -sfn /opt/openhabian/openhabian-setup.sh /usr/local/bin/openhabian-config
 
 # shellcheck disable=SC2154
 echo "$(timestamp) [openHABian] Executing openhabian-setup.sh unattended... "
-set +x
 if (/bin/bash /opt/openhabian/openhabian-setup.sh unattended); then
   rm -f /opt/openHABian-install-inprogress
   touch /opt/openHABian-install-successful

--- a/build-image/first-boot.bash
+++ b/build-image/first-boot.bash
@@ -160,12 +160,14 @@ if apt-get install --yes git &>/dev/null; then echo "OK"; else echo "FAILED"; fa
 
 if [ -d /opt/openhabian/ ]; then cd /opt && rm -rf /opt/openhabian/; fi
 # shellcheck disable=SC2154
+set -x
 echo "$(timestamp) [openHABian] Cloning myself from ${repositoryurl}, ${clonebranch} branch... "
 if git clone -q -b "$clonebranch" "$repositoryurl" /opt/openhabian; then echo "OK"; else echo "FAILED"; fail_inprogress; fi
 ln -sfn /opt/openhabian/openhabian-setup.sh /usr/local/bin/openhabian-config
 
 # shellcheck disable=SC2154
 echo "$(timestamp) [openHABian] Executing openhabian-setup.sh unattended... "
+set +x
 if (/bin/bash /opt/openhabian/openhabian-setup.sh unattended); then
   rm -f /opt/openHABian-install-inprogress
   touch /opt/openHABian-install-successful

--- a/functions/helpers.bash
+++ b/functions/helpers.bash
@@ -32,6 +32,28 @@ cond_echo() {
   fi
 }
 
+## Add keys to apt for new package sources
+## Valid Arguments: URL
+##
+##    add_keys(String url)
+##
+add_keys() {
+	  local repoKey
+	  
+  repoKey="$(mktemp /tmp/openhabian.XXXXX)"
+  
+  echo -n "$(timestamp) [openHABian] Adding required keys to apt... "
+  cond_redirect wget -qO "$repoKey" "$1"
+  if cond_redirect apt-key add "$repoKey"; then
+    echo "OK"
+    rm -f "$repoKey"
+  else
+    echo "FAILED"
+    rm -f "$repoKey"
+    return 1;
+  fi
+}
+
 # fingerprinting based on
 # https://www.raspberrypi.org/documentation/hardware/raspberrypi/revision-codes/README.md
 is_pizero() {

--- a/functions/helpers.bash
+++ b/functions/helpers.bash
@@ -225,7 +225,7 @@ tryUntil() {
   local i=$count
   interval=${3:-1}
   until [ "$i" -le 0 ]; do
-    cond_echo -n "(executing ${cmd})"
+    cond_echo "(executing ${cmd})\c"
     eval "${cmd}"
     ret=$?
     if [ $ret -eq 0 ]; then break; fi

--- a/functions/helpers.bash
+++ b/functions/helpers.bash
@@ -225,7 +225,7 @@ tryUntil() {
   local i=$count
   interval=${3:-1}
   until [ "$i" -le 0 ]; do
-    cond_echo "(executing ${cmd})\c"
+    cond_echo "\n(executing ${cmd})\c"
     eval "${cmd}"
     ret=$?
     if [ $ret -eq 0 ]; then break; fi

--- a/functions/helpers.bash
+++ b/functions/helpers.bash
@@ -225,7 +225,7 @@ tryUntil() {
   local i=$count
   interval=${3:-1}
   until [ "$i" -le 0 ]; do
-    cond_echo "(executing ${cmd}) \c"
+    cond_echo "(executing ${cmd}) ${COL_DEF}\c"
     eval "${cmd}"
     ret=$?
     if [ $ret -eq 0 ]; then break; fi

--- a/functions/helpers.bash
+++ b/functions/helpers.bash
@@ -225,7 +225,7 @@ tryUntil() {
   local i=$count
   interval=${3:-1}
   until [ "$i" -le 0 ]; do
-    cond_echo "\n(executing ${cmd})\c"
+    cond_echo "(executing ${cmd}) \c"
     eval "${cmd}"
     ret=$?
     if [ $ret -eq 0 ]; then break; fi

--- a/functions/menu.bash
+++ b/functions/menu.bash
@@ -233,9 +233,6 @@ show_main_menu() {
     if [[ $choosenComponents == *"6A"* ]]; then init_zram_mounts install; fi
     if [[ $choosenComponents == *"Uninstall ZRAM"* ]]; then init_zram_mounts remove; fi
 
-  elif [[ "$choice" == "99"* ]]; then
-    show_about
-
   else
     whiptail --msgbox "Error: unrecognized option \"$choice\"" 10 60
   fi

--- a/functions/menu.bash
+++ b/functions/menu.bash
@@ -26,7 +26,7 @@ Menu 60 finally is a shortcut to offer all option for (un)installation in a sing
 }
 
 show_main_menu() {
-  choice=$(whiptail --title "Welcome to the openHABian Configuration Tool $(get_git_revision)" --menu "Setup Options" 21 116 14 --cancel-button Exit --ok-button Execute \
+  choice=$(whiptail --title "Welcome to the openHABian Configuration Tool $(get_git_revision)" --menu "Setup Options" 20 116 13 --cancel-button Exit --ok-button Execute \
   "00 | About openHABian"        "Information about the openHABian project and this tool" \
   "" "" \
   "01 | Select Branch"           "Select the openHABian config tool version (\"branch\") to run" \
@@ -38,9 +38,8 @@ show_main_menu() {
   "30 | System Settings"         "A range of system and hardware related configuration steps ►" \
   "40 | openHAB related"         "Switch the installed openHAB version or apply tweaks ►" \
   "50 | Backup/Restore"          "Manage backups and restore your system ►" \
-  "60 | Manual/Fresh Setup"      "Go through all openHABian setup steps manually ►" \
   "" "" \
-  "99 | Help"                    "Further options and guidance with Linux and openHAB" \
+  "60 | Manual/Fresh Setup"      "Go through all openHABian setup steps manually ►" \
   3>&1 1>&2 2>&3)
   RET=$?
   if [ $RET -eq 1 ] || [ $RET -eq 255 ]; then

--- a/functions/nodejs-apps.bash
+++ b/functions/nodejs-apps.bash
@@ -1,7 +1,5 @@
 #!/usr/bin/env bash
 
-timestamp() { date +"%F_%T_%Z"; }
-
 nodejs_setup() {
   if command -v npm &>/dev/null; then
     return 0
@@ -47,7 +45,6 @@ frontail_setup() {
   cp "${BASEDIR}"/includes/frontail-theme.css "$frontail_base"/web/assets/styles/openhab.css
   sed -e "s|%FRONTAILBASE|${frontail_base}|g" "${BASEDIR}"/includes/frontail.service > /etc/systemd/system/frontail.service
   chmod 664 /etc/systemd/system/frontail.service
-  cond_redirect systemctl daemon-reload
   cond_redirect systemctl enable frontail.service
   if cond_redirect systemctl restart frontail; then echo "OK"; else echo "FAILED (service)"; return 1; fi
   if ! running_in_docker; then

--- a/functions/nodejs-apps.bash
+++ b/functions/nodejs-apps.bash
@@ -50,7 +50,9 @@ frontail_setup() {
   cond_redirect systemctl daemon-reload
   cond_redirect systemctl enable frontail.service
   if cond_redirect systemctl restart frontail; then echo "OK"; else echo "FAILED (service)"; return 1; fi
-  dashboard_add_tile frontail
+  if ! running_in_docker; then
+    dashboard_add_tile frontail
+  fi
 }
 
 nodered_setup() {

--- a/functions/nodejs-apps.bats
+++ b/functions/nodejs-apps.bats
@@ -9,7 +9,7 @@ setup_file() {
 
 @test "installation-Frontail_is_running" {
 run frontail_setup
-if [ "$status" -ne 0 ]; then echo "$output" >&3; fi
+#if [ "$status" -ne 0 ]; then echo "$output" >&3; fi
 run frontail -V
 if [ "$status" -ne 0 ]; then echo "$output" >&3; fi
 [ "$status" -eq 0 ]

--- a/functions/nodejs-apps.bats
+++ b/functions/nodejs-apps.bats
@@ -9,7 +9,7 @@ setup_file() {
 
 @test "installation-Frontail_is_running" {
 run frontail_setup
-#if [ "$status" -ne 0 ]; then echo "$output" >&3; fi
+if [ "$status" -ne 0 ]; then echo "$output" >&3; fi
 run frontail -V
 if [ "$status" -ne 0 ]; then echo "$output" >&3; fi
 [ "$status" -eq 0 ]

--- a/functions/nodejs-apps.bats
+++ b/functions/nodejs-apps.bats
@@ -7,12 +7,19 @@ setup_file() {
   export BASEDIR="${BATS_TEST_DIRNAME}/.."
 }
 
-@test "installation-Frontail_is_running" {
-run frontail_setup
-#if [ "$status" -ne 0 ]; then echo "$output" >&3; fi
-run frontail -V
-if [ "$status" -ne 0 ]; then echo "$output" >&3; fi
-[ "$status" -eq 0 ]
+teardown_file() {
+  unset BASEDIR
+  systemctl stop frontail.service || true
+}
 
-echo -e "# \e[32mFrontail properly installed." >&3
+@test "installation-frontail_install" {
+  echo -e "# \e[36mFrontail installation starting...\e[0m" >&3
+  run frontail_setup
+  if [ "$status" -ne 0 ]; then echo "$output" >&3; fi
+  [ "$status" -eq 0 ]
+  echo -e "# \e[32mFrontail installation successful.\e[0m" >&3
+  run systemctl is-active --quiet frontail.service
+  if [ "$status" -ne 0 ]; then echo "$output" >&3; fi
+  [ "$status" -eq 0 ]
+  echo -e "# \e[32mFrontail service is running.\e[0m" >&3
 }

--- a/functions/packages.bash
+++ b/functions/packages.bash
@@ -102,30 +102,31 @@ etckeeper_setup() {
 }
 
 homegear_setup() {
-  FAILED=0
-  introtext="This will install Homegear, the Homematic CCU2 emulation software, in the latest stable release from the official repository."
-  failtext="Sadly there was a problem setting up the selected option. Please report this problem in the openHAB community forum or as a openHABian GitHub issue."
-  successtext="Setup was successful.
-Homegear is now up and running. Next you might want to edit the configuration file '/etc/homegear/families/homematicbidcos.conf' or adopt devices through the homegear console, reachable by 'sudo homegear -r'.
-Please read up on the homegear documentation for more details: https://doc.homegear.eu/data/homegear
-To continue your integration in openHAB 2, please follow the instructions under: https://www.openhab.org/addons/bindings/homematic/
-"
+  local distro
+  local introtext
+  local successtext
 
-  echo -n "$(timestamp) [openHABian] Setting up the Homematic CCU2 emulation software Homegear... "
-  if is_pine64; then
-    if [ -n "$INTERACTIVE" ]; then
-      whiptail --title "Incompatible Hardware Detected" --msgbox "We are sorry, Homegear is not yet available for your platform." 10 60
-    fi
-    echo "FAILED (incompatible)"; return 1
-  fi
-
-  if [ -n "$INTERACTIVE" ]; then
-    if ! (whiptail --title "Description, Continue?" --yes-button "Continue" --no-button "Back" --yesno "$introtext" 15 80); then echo "CANCELED"; return 0; fi
-  fi
-
-  cond_redirect wget -O - http://homegear.eu/packages/Release.key | apt-key add -
+  echo -n "$(timestamp) [openHABian] Installing Homegear required packages... "
+  if cond_redirect apt-get install --yes lsb-release; then echo "OK"; else echo "FAILED"; return 1; fi
 
   distro="$(lsb_release -si)-$(lsb_release -sc)"
+  introtext="This will install Homegear, the Homematic CCU2 emulation software, using the latest stable release available from the official repository."
+  successtext="Setup was successful.\\n\\nHomegear is now up and running. Next you might want to edit the configuration file '/etc/homegear/families/homematicbidcos.conf' or adopt devices through the homegear console, reachable by 'sudo homegear -r'.\\n\\nPlease read up on the homegear documentation for more details: https://doc.homegear.eu/data/homegear\\n\\nTo continue your integration in openHAB 2, please follow the instructions under: https://www.openhab.org/addons/bindings/homematic/"
+
+  echo -n "$(timestamp) [openHABian] Beginning Homematic CCU2 emulation software Homegear install... "
+  if is_pine64; then
+    if [ -n "$INTERACTIVE" ]; then
+      whiptail --title "Incompatible Hardware Detected!" --msgbox "We are sorry, Homegear is not available for your platform at this time." 8 60
+    fi
+    echo "FAILED (incompatible)"
+    return 1
+  fi
+  if [ -n "$INTERACTIVE" ]; then
+    if ! (whiptail --title "Homegear installation?" --yes-button "Continue" --no-button "Cancel" --yesno "$introtext" 8 80); then echo "CANCELED"; return 0; fi
+  fi
+
+  if ! add_keys "https://apt.homegear.eu/Release.key"; then return 1; fi
+
   case "$distro" in
     Debian-jessie)
       echo 'deb https://apt.homegear.eu/Debian/ jessie/' > /etc/apt/sources.list.d/homegear.list
@@ -135,12 +136,6 @@ To continue your integration in openHAB 2, please follow the instructions under:
       ;;
     Debian-buster)
       echo 'deb https://apt.homegear.eu/Debian/ buster/' > /etc/apt/sources.list.d/homegear.list
-      ;;
-    Debian-bullseye)
-      echo 'deb https://apt.homegear.eu/Debian/ bullseye/' > /etc/apt/sources.list.d/homegear.list
-      ;;
-    Debian-sid)
-      echo 'deb https://apt.homegear.eu/Debian/ sid/' > /etc/apt/sources.list.d/homegear.list
       ;;
     Raspbian-jessie)
       echo 'deb https://apt.homegear.eu/Raspbian/ jessie/' > /etc/apt/sources.list.d/homegear.list
@@ -153,9 +148,6 @@ To continue your integration in openHAB 2, please follow the instructions under:
       ;;
     Raspbian-bullseye)
       echo 'deb https://apt.homegear.eu/Raspbian/ bullseye/' > /etc/apt/sources.list.d/homegear.list
-      ;;
-    Raspbian-sid)
-      echo 'deb https://apt.homegear.eu/Raspbian/ sid/' > /etc/apt/sources.list.d/homegear.list
       ;;
     Ubuntu-trusty)
       echo 'deb https://apt.homegear.eu/Ubuntu/ trusty/' > /etc/apt/sources.list.d/homegear.list
@@ -175,23 +167,23 @@ To continue your integration in openHAB 2, please follow the instructions under:
       ;;
   esac
 
+  echo -n "$(timestamp) [openHABian] Installing Homegear... "
   if ! cond_redirect apt-get update; then echo "FAILED"; return 1; fi
-  if ! cond_redirect apt-get install --yes homegear homegear-homematicbidcos homegear-homematicwired; then echo "FAILED"; return 1; fi
-  cond_redirect systemctl enable homegear.service
-  cond_redirect systemctl start homegear.service
-  cond_redirect adduser "${username:-openhabian}" homegear
-  cond_redirect adduser openhab homegear
-  echo "OK"
+  if cond_redirect apt-get install --yes homegear homegear-homematicbidcos homegear-homematicwired; then echo "OK"; else echo "FAILED"; return 1; fi
+
+  echo -n "$(timestamp) [openHABian] Setting up Homegear user account permisions... "
+  if ! cond_redirect adduser "${username:-openhabian}" homegear; then echo "FAILED"; return 1; fi
+  if cond_redirect adduser openhab homegear; then echo "OK"; else echo "FAILED"; return 1; fi
+
+  echo -n "$(timestamp) [openHABian] Setting up Homegear service... "
+  cp "$BASEDIR"/includes/homegear.service /lib/systemd/system/homegear.service
+  if ! systemctl enable homegear.service; then echo "FAILED"; return 1; fi
+  if systemctl restart homegear.service; then echo "OK"; else echo "FAILED"; return 1; fi
 
   if [ -n "$INTERACTIVE" ]; then
-    if [ "$FAILED" -eq 0 ]; then
-      whiptail --title "Operation Successful!" --msgbox "$successtext" 15 80
-    else
-      whiptail --title "Operation Failed!" --msgbox "$failtext" 10 60
-    fi
+    whiptail --title "Operation Successful!" --msgbox "$successtext" 14 80
   fi
 }
-
 mqtt_setup() {
   FAILED=0
   introtext="The MQTT broker Eclipse Mosquitto will be installed through the official repository, as desribed in: https://mosquitto.org/2013/01/mosquitto-debian-repository\\nAdditionally you can activate username:password authentication.\\n\\nHEADS UP: Only proceed when you are aware that this will be in conflict with use of the MQTTv2 binding which will also be using the same ports."

--- a/functions/packages.bash
+++ b/functions/packages.bash
@@ -102,15 +102,17 @@ etckeeper_setup() {
 }
 
 homegear_setup() {
-  local distro
   local introtext
+  local myOS
+  local myRelease
   local successtext
 
   echo -n "$(timestamp) [openHABian] Installing Homegear required packages... "
   if cond_redirect apt-get install --yes lsb-release; then echo "OK"; else echo "FAILED"; return 1; fi
 
-  distro="$(lsb_release -si)-$(lsb_release -sc)"
   introtext="This will install Homegear, the Homematic CCU2 emulation software, using the latest stable release available from the official repository."
+  myOS="$(lsb_release -si)"
+  myRelease="$(lsb_release -sc)"
   successtext="Setup was successful.\\n\\nHomegear is now up and running. Next you might want to edit the configuration file '/etc/homegear/families/homematicbidcos.conf' or adopt devices through the homegear console, reachable by 'sudo homegear -r'.\\n\\nPlease read up on the homegear documentation for more details: https://doc.homegear.eu/data/homegear\\n\\nTo continue your integration in openHAB 2, please follow the instructions under: https://www.openhab.org/addons/bindings/homematic/"
 
   echo -n "$(timestamp) [openHABian] Beginning Homematic CCU2 emulation software Homegear install... "
@@ -127,9 +129,7 @@ homegear_setup() {
 
   if ! add_keys "https://apt.homegear.eu/Release.key"; then return 1; fi
 
-  myOS=$(echo $distro | cut -d'-' -f1)
-  myRelease=$(echo $distro | cut -d'-' -f2)
-  echo 'deb https://apt.homegear.eu/${myOS}/ ${myRelease}/' > /etc/apt/sources.list.d/homegear.list
+  echo "deb https://apt.homegear.eu/${myOS}/ ${myRelease}/" > /etc/apt/sources.list.d/homegear.list
 
   echo -n "$(timestamp) [openHABian] Installing Homegear... "
   if ! cond_redirect apt-get update; then echo "FAILED"; return 1; fi
@@ -148,6 +148,7 @@ homegear_setup() {
     whiptail --title "Operation Successful!" --msgbox "$successtext" 14 80
   fi
 }
+
 mqtt_setup() {
   FAILED=0
   introtext="The MQTT broker Eclipse Mosquitto will be installed through the official repository, as desribed in: https://mosquitto.org/2013/01/mosquitto-debian-repository\\nAdditionally you can activate username:password authentication.\\n\\nHEADS UP: Only proceed when you are aware that this will be in conflict with use of the MQTTv2 binding which will also be using the same ports."

--- a/functions/packages.bash
+++ b/functions/packages.bash
@@ -127,54 +127,9 @@ homegear_setup() {
 
   if ! add_keys "https://apt.homegear.eu/Release.key"; then return 1; fi
 
-  case "$distro" in
-    Debian-jessie)
-      echo 'deb https://apt.homegear.eu/Debian/ jessie/' > /etc/apt/sources.list.d/homegear.list
-      ;;
-    Debian-stretch)
-      echo 'deb https://apt.homegear.eu/Debian/ stretch/' > /etc/apt/sources.list.d/homegear.list
-      ;;
-    Debian-buster)
-      echo 'deb https://apt.homegear.eu/Debian/ buster/' > /etc/apt/sources.list.d/homegear.list
-      ;;
-    Debian-bullseye)
-      echo 'deb https://apt.homegear.eu/Debian/ bullseye/' > /etc/apt/sources.list.d/homegear.list
-      ;;
-    Debian-sid)
-      echo 'deb https://apt.homegear.eu/Debian/ sid/' > /etc/apt/sources.list.d/homegear.list
-      ;;
-    Raspbian-jessie)
-      echo 'deb https://apt.homegear.eu/Raspbian/ jessie/' > /etc/apt/sources.list.d/homegear.list
-      ;;
-    Raspbian-stretch)
-      echo 'deb https://apt.homegear.eu/Raspbian/ stretch/' > /etc/apt/sources.list.d/homegear.list
-      ;;
-    Raspbian-buster)
-      echo 'deb https://apt.homegear.eu/Raspbian/ buster/' > /etc/apt/sources.list.d/homegear.list
-      ;;
-    Raspbian-bullseye)
-      echo 'deb https://apt.homegear.eu/Raspbian/ bullseye/' > /etc/apt/sources.list.d/homegear.list
-      ;;
-    Raspbian-sid)
-      echo 'deb https://apt.homegear.eu/Raspbian/ sid/' > /etc/apt/sources.list.d/homegear.list
-      ;;
-    Ubuntu-trusty)
-      echo 'deb https://apt.homegear.eu/Ubuntu/ trusty/' > /etc/apt/sources.list.d/homegear.list
-      ;;
-    Ubuntu-xenial)
-      echo 'deb https://apt.homegear.eu/Ubuntu/ xenial/' > /etc/apt/sources.list.d/homegear.list
-      ;;
-    Ubuntu-bionic)
-      echo 'deb https://apt.homegear.eu/Ubuntu/ bionic/' > /etc/apt/sources.list.d/homegear.list
-      ;;
-    Ubuntu-focal)
-      echo 'deb https://apt.homegear.eu/Ubuntu/ focal/' > /etc/apt/sources.list.d/homegear.list
-      ;;
-    *)
-      cond_echo "Sorry, your OS is not supported."
-      return 1
-      ;;
-  esac
+  myOS=$(echo $distro | cut -d'-' -f1)
+  myRelease=$(echo $distro | cut -d'-' -f2)
+  echo 'deb https://apt.homegear.eu/${myOS}/ ${myRelease}/' > /etc/apt/sources.list.d/homegear.list
 
   echo -n "$(timestamp) [openHABian] Installing Homegear... "
   if ! cond_redirect apt-get update; then echo "FAILED"; return 1; fi

--- a/functions/packages.bash
+++ b/functions/packages.bash
@@ -137,6 +137,12 @@ homegear_setup() {
     Debian-buster)
       echo 'deb https://apt.homegear.eu/Debian/ buster/' > /etc/apt/sources.list.d/homegear.list
       ;;
+    Debian-bullseye)
+      echo 'deb https://apt.homegear.eu/Debian/ bullseye/' > /etc/apt/sources.list.d/homegear.list
+      ;;
+    Debian-sid)
+      echo 'deb https://apt.homegear.eu/Debian/ sid/' > /etc/apt/sources.list.d/homegear.list
+      ;;
     Raspbian-jessie)
       echo 'deb https://apt.homegear.eu/Raspbian/ jessie/' > /etc/apt/sources.list.d/homegear.list
       ;;
@@ -148,6 +154,9 @@ homegear_setup() {
       ;;
     Raspbian-bullseye)
       echo 'deb https://apt.homegear.eu/Raspbian/ bullseye/' > /etc/apt/sources.list.d/homegear.list
+      ;;
+    Raspbian-sid)
+      echo 'deb https://apt.homegear.eu/Raspbian/ sid/' > /etc/apt/sources.list.d/homegear.list
       ;;
     Ubuntu-trusty)
       echo 'deb https://apt.homegear.eu/Ubuntu/ trusty/' > /etc/apt/sources.list.d/homegear.list

--- a/functions/packages.bats
+++ b/functions/packages.bats
@@ -15,7 +15,8 @@ teardown_file() {
 
 @test "destructive-homegear_install" {
   echo -e "# \e[36mHomegear installation starting..." >&3
-  run homegear_setup
+  # TODO: fix homegear installation, currently it does not finish
+  #run homegear_setup
   if [ "$status" -ne 0 ]; then echo "$output" >&3; fi
   [ "$status" -eq 0 ]
   echo -e "# \e[32mHomegear installation successful." >&3

--- a/functions/packages.bats
+++ b/functions/packages.bats
@@ -3,30 +3,36 @@
 load packages
 load helpers
 
+setup_file() {
+  export BASEDIR="${BATS_TEST_DIRNAME}/.."
+}
 
 teardown_file() {
-  echo -e "# \e[36mHomegear test cleanup..." >&3
+  unset BASEDIR
   systemctl stop homegear.service || true
-  # TODO: fix homegear shutdown handling, is does not properly shut down and will cause bats tests to hang
-  #ps aux|grep -i homeg >&3
-  # workaround in place:
-  killall -9 homegear || true
+  systemctl stop mosquitto.service || true
 }
 
 @test "destructive-homegear_install" {
-  echo -e "# \e[36mHomegear installation starting..." >&3
-  # TODO: fix homegear installation, currently it does not finish
-  #run homegear_setup
+  echo -e "# \e[36mHomegear installation starting...\e[0m" >&3
+  run homegear_setup
   if [ "$status" -ne 0 ]; then echo "$output" >&3; fi
   [ "$status" -eq 0 ]
-  echo -e "# \e[32mHomegear installation successful." >&3
-  # TODO: fix homegear service, currently it does not detect running instance in CI
-  #run systemctl status homegear.service
-  #echo "$output" >&3
-  #run systemctl is-active --quiet homegear.service
-  #if [ "$status" -ne 0 ]; then echo "$output" >&3; fi
-  # workaround in place: (pgrep would be better, but in doubt that it is installed in CI)
-  run bash -c "ps aux | grep homegear | grep -v grep"
+  echo -e "# \e[32mHomegear installation successful.\e[0m" >&3
+  run systemctl is-active --quiet homegear.service
+  if [ "$status" -ne 0 ]; then echo "$output" >&3; fi
   [ "$status" -eq 0 ]
-  echo -e "# \e[32mHomegear service running." >&3
+  echo -e "# \e[32mHomegear service is running.\e[0m" >&3
+}
+
+@test "destructive-mqtt_install" {
+  echo -e "# \e[36mMQTT installation starting...\e[0m" >&3
+  run mqtt_setup
+  if [ "$status" -ne 0 ]; then echo "$output" >&3; fi
+  [ "$status" -eq 0 ]
+  echo -e "# \e[32mMQTT installation successful.\e[0m" >&3
+  run systemctl is-active --quiet mosquitto.service
+  if [ "$status" -ne 0 ]; then echo "$output" >&3; fi
+  [ "$status" -eq 0 ]
+  echo -e "# \e[32mMQTT service is running.\e[0m" >&3
 }

--- a/functions/system.bash
+++ b/functions/system.bash
@@ -143,8 +143,8 @@ create_mount() {
 
 srv_bind_mounts() {
   echo -n "$(timestamp) [openHABian] Preparing openHAB folder mounts under /srv/... "
-  systemctl is-active --quiet smbd >/dev/null 2>&1 && systemctl stop smbd
-  systemctl is-active --quiet zram-config >/dev/null 2>&1 && systemctl stop zram-config
+  cond_redirect systemctl is-active --quiet smbd && systemctl stop smbd
+  cond_redirect systemctl is-active --quiet zram-config && systemctl stop zram-config
   cond_redirect umount -q /srv/openhab2-{sys,conf,userdata,logs,addons}
   cond_redirect rm -f /etc/systemd/system/srv*.mount
   cond_redirect mkdir -p /srv/openhab2-{sys,conf,userdata,logs,addons}

--- a/functions/system.bash
+++ b/functions/system.bash
@@ -146,8 +146,9 @@ srv_bind_mounts() {
   systemctl is-active --quiet smbd >/dev/null 2>&1 && systemctl stop smbd
   systemctl is-active --quiet zram-config >/dev/null 2>&1 && systemctl stop zram-config
   cond_redirect umount -q /srv/openhab2-{sys,conf,userdata,logs,addons}
-  sed -i "\\#[ \\t]/srv/openhab2-#d" /etc/fstab
+#  sed -i "\\#[ \\t]/srv/openhab2-#d" /etc/fstab
   # dummy " to fix vim coloring
+  cat /etc/fstab
   cond_redirect rm -f /etc/systemd/system/srv*.mount
   cond_redirect mkdir -p /srv/openhab2-{sys,conf,userdata,logs,addons}
   cond_redirect cp "$BASEDIR"/includes/srv_readme.txt /srv/README.txt
@@ -161,6 +162,7 @@ srv_bind_mounts() {
 
   if [ -f /etc/ztab ]; then systemctl start zram-config; fi
   if [ -f /etc/samba/smb.conf ]; then systemctl start smbd; fi
+  echo "OK"
 }
 
 permissions_corrections() {

--- a/functions/system.bash
+++ b/functions/system.bash
@@ -143,9 +143,8 @@ create_mount() {
 
 srv_bind_mounts() {
   echo -n "$(timestamp) [openHABian] Preparing openHAB folder mounts under /srv/... "
-set -x
-  systemctl is-active --quiet smbd && systemctl stop smbd
-  systemctl is-active --quiet zram-config && systemctl stop zram-config
+  systemctl is-active --quiet smbd >/dev/null 2>&1 && systemctl stop smbd
+  systemctl is-active --quiet zram-config >/dev/null 2>&1 && systemctl stop zram-config
   cond_redirect umount -q /srv/openhab2-{sys,conf,userdata,logs,addons}
   sed -i "\\#[ \\t]/srv/openhab2-#d" /etc/fstab
   # dummy " to fix vim coloring

--- a/functions/system.bash
+++ b/functions/system.bash
@@ -146,9 +146,6 @@ srv_bind_mounts() {
   systemctl is-active --quiet smbd >/dev/null 2>&1 && systemctl stop smbd
   systemctl is-active --quiet zram-config >/dev/null 2>&1 && systemctl stop zram-config
   cond_redirect umount -q /srv/openhab2-{sys,conf,userdata,logs,addons}
-#  sed -i "\\#[ \\t]/srv/openhab2-#d" /etc/fstab
-  # dummy " to fix vim coloring
-  cat /etc/fstab
   cond_redirect rm -f /etc/systemd/system/srv*.mount
   cond_redirect mkdir -p /srv/openhab2-{sys,conf,userdata,logs,addons}
   cond_redirect cp "$BASEDIR"/includes/srv_readme.txt /srv/README.txt

--- a/functions/system.bash
+++ b/functions/system.bash
@@ -143,6 +143,7 @@ create_mount() {
 
 srv_bind_mounts() {
   echo -n "$(timestamp) [openHABian] Preparing openHAB folder mounts under /srv/... "
+set -x
   systemctl is-active --quiet smbd && systemctl stop smbd
   systemctl is-active --quiet zram-config && systemctl stop zram-config
   cond_redirect umount -q /srv/openhab2-{sys,conf,userdata,logs,addons}

--- a/functions/zram.bash
+++ b/functions/zram.bash
@@ -35,14 +35,14 @@ zram_setup() {
   if is_arm; then
     if ! has_lowmem && ! is_pione && ! is_cmone && ! is_pizero && ! is_pizerow; then
       cond_redirect systemctl stop openhab2
-      echo -n "$(timestamp) [openHABian] Installing ZRAM ..."
+      echo "$(timestamp) [openHABian] Installing ZRAM ..."
       init_zram_mounts install
       cond_redirect systemctl start openhab2
     else
-      echo -n "$(timestamp) [openHABian] Skipping ZRAM install on ARM hardware without enough memory."
+      echo "$(timestamp) [openHABian] Skipping ZRAM install on ARM hardware without enough memory."
     fi
   else
-    echo -n "$(timestamp) [openHABian] Skipping ZRAM install on non-ARM hardware."
+    echo "$(timestamp) [openHABian] Skipping ZRAM install on non-ARM hardware."
   fi
 }
 

--- a/functions/zram.bash
+++ b/functions/zram.bash
@@ -36,7 +36,7 @@ zram_setup() {
     if ! has_lowmem && ! is_pione && ! is_cmone && ! is_pizero && ! is_pizerow; then
       cond_redirect systemctl stop openhab2
       echo "$(timestamp) [openHABian] Installing ZRAM ..."
-      init_zram_mounts install
+      cond_redirect init_zram_mounts install
       cond_redirect systemctl start openhab2
     else
       echo "$(timestamp) [openHABian] Skipping ZRAM install on ARM hardware without enough memory."

--- a/includes/homegear.service
+++ b/includes/homegear.service
@@ -1,0 +1,11 @@
+[Unit]
+Description=Homegear
+
+[Service]
+Type=simple
+User=homegear
+Group=homegear
+ExecStart=/usr/bin/homegear
+
+[Install]
+WantedBy=multi-user.target

